tests/functional/test_s3.py

```diff
@@ -3702,10 +3737,12 @@ def _verify_presigned_url_addressing(
 
 
 class TestS3XMLPayloadEscape(BaseS3OperationTest):
-    def assert_correct_content_md5(self, request):
-        content_md5_bytes = get_md5(request.body).digest()
-        content_md5 = base64.b64encode(content_md5_bytes)
-        self.assertEqual(content_md5, request.headers["Content-MD5"])
+    def assert_correct_crc32_checksum(self, request):
+        checksum = CrtCrc32Checksum() if HAS_CRT else Crc32Checksum()
```

Comment:
Should we be surfacing `_CHECKSUM_CLASS` in some way to do this automatically. Otherwise, we're desyncing the code from the test.

Reply:
Yep, I don't think we should be importing the private variable directly in code. For this test harness since it's not library code, we may establish a codified reliance on this even those it's private. That way the test fails on removal and it's clear what we're doing.

```python
import botocore.httpchecksum
[...]
def get_default_checksum_cls():
    # This pass through is grabbing our internally supported list of checksums
    # to ensure we stay in sync, while not exposing them publicly.
    return botocore.httpchecksum._CHECKSUM_CLS["crc32"]
```